frontend/src/modules/TournamentView/view.js

```diff
@@ -14,12 +14,18 @@ class TournamentView extends React.Component {
     };
   }
   componentDidMount() {
-    console.log(this.props);
     const { tournament } = this.state;
     if (!tournament) {
-      console.log(this.props.match.params.id);
       fetch(`http://localhost:8000/tournaments/${this.props.match.params.id}`)
-        .then(res => res.json())
+        .then(res => {
+          if (!res.ok) {
+            alert(
+              'Sorry this tournament does not exist. Click Ok to redirect to the home page'
+            );
+            window.location = 'http://localhost:3000/tournaments/';
```

Comment:
This is the right idea but when this code is running in production the path won't be localhost:3000. The cleaner way is to use the `history` prop that react-router provides us. Instead of `window.location = 'http://localhost:3000/tournaments/';` we can do `this.props.history.push('/tournaments');`